woocommerce/woocommerce-admin
# Calendar: Use isInvalidDate to disallow date selections

Introduce `isInvalidDate` to `DatePicker` and `DateRange` now that Gutenberg allows invalid days on the single date picker.

In an effort to make the two APIs identical, I removed the `{ invalidDays="future" }` helper prop/function because it just masks functionality by adding complexity. I'd much rather expose the underlying ability to suit everyone's needs.

This is a breaking change, but we are still on the first version so I _think_ its ok not to bump to `2.0.0`. I'm open to bumping though.

### Screenshots

![screen shot 2019-02-25 at 12 37 28 pm](https://user-images.githubusercontent.com/1922453/53307131-28437600-38fa-11e9-812d-bd245b9dfee2.png)

### Detailed test instructions:


1. Use master branch of Gutenberg or 7.1.0 of the `components` package. 
1. Customers Report.
2. Show > Advanced Filters > Registered.
3. See that future dates in the date picker are invalid and can't be selected.
4. Orders Report > Date Range > Custom.
5. Again, future dates should be invalidated.
